src/components/InventoryGroupDetail/GroupTabDetails.js

```diff
@@ -24,12 +25,44 @@ const GroupTabDetailsWrapper = ({
   hasEdgeImages,
 }) => {
   const [tab, setTab] = useState(0);
+
   const { hasAccess: canViewHosts } = usePermissionsWithContext(
     REQUIRED_PERMISSIONS_TO_READ_GROUP_HOSTS(groupId)
   );
+  const conventionalSystemsContent = (
+    <GroupSystems
+      groupName={groupName}
+      groupId={groupId}
+      hostType={hybridInventoryTabKeys.conventional.key}
+    />
+  );
+
+  const immutableSystemsContent = (
+    <GroupImmutableSystems
+      groupId={groupId}
+      groupName={groupName}
+      hostType={hybridInventoryTabKeys.immutable.key}
+    />
+  );
+
+  const [component, setComponent] = useState(conventionalSystemsContent);
 
+  useEffect(() => {
+    setComponent(null);
+    if (activeTab == hybridInventoryTabKeys.conventional.key) {
+      setComponent(conventionalSystemsContent);
+    } else {
+      setComponent(immutableSystemsContent);
+    }
+  }, [activeTab]);
   const handleTabClick = (_event, tabIndex) => {
+    setComponent(null);
     setTab(tabIndex);
+    if (tabIndex == hybridInventoryTabKeys.conventional.key) {
+      setComponent(conventionalSystemsContent);
+    } else {
+      setComponent(immutableSystemsContent);
+    }
   };
 
   const [activeTabKey, setActiveTabKey] = useState(0);
```

Comment:
It feels like we duplicating the "active tab" logic here, whereas it is already implemented in a higher level component here https://github.com/RedHatInsights/insights-inventory-frontend/blob/master/src/components/InventoryGroupDetail/InventoryGroupDetail.js#L50. It feels redundant/a lot of duplication going here. Can we reduce the tabs handling logic to only one component? Otherwise, this gets hard to understand and especially maintain very quick.

Reply:
@gkarat this screen has 2 components with tabs [Systems | Group Info] handled by ActiveTabKey and [Conventional | Immutable] that uses ActiveTab
the context between this 2 levels are different and lives on on details and the other on the tabs